doc/guides/modules.rst

```diff
@@ -18,6 +18,19 @@ Zephyr depends on several categories of modules, including:
 - File Systems
 - Inter-Process Communication (IPC) libraries
 
+Zephyr also supports extension of the functionaly with out-of-tree modules.
+Out-of-tree modules will likely be their own manifest repository during
```

Comment:
This should link to the west documentation regarding topologies, so that users are informed of what "be their own manifest repository" means